mg/deps.go

```diff
@@ -130,6 +130,9 @@ func runDeps(ctx context.Context, fns []Fn) {
 func checkFns(fns []interface{}) []Fn {
 	funcs := make([]Fn, len(fns))
 	for i, f := range fns {
+		if f == nil {
+			panic("A dependency of the current target was defined improperly, with parenthesis. Dependencies should be defined as mg.Deps(MyDep), not mg.Deps(MyDep())")
```

Comment:
is panic the right thing here, or can we log and `os.Exit(1)` or similar?

Reply:
I used panic here because mage has a recover handler for panics that prints the error and exits with a non-zero code.

I just tried this branch again after the latest changes from main were merged and my patch doesn't seem to be working anymore though (it just returns the nil pointer instead now... not sure what changed to break the patch). So I'll have to take a look this weekend.

Were you seeing this panic not behave correctly that prompted you to add this comment? Just wondering if you are seeing different behavior with this patch than I am.